src/mutation.cpp

```diff
@@ -40,40 +44,19 @@ void Character::toggle_trait(const std::string &flag)
 
 void Character::toggle_mutation(const std::string &flag)
 {
-    toggle_str_set(my_mutations, flag); //Toggles a mutation on the player
-    recalc_sight_limits();
-}
-
-void Character::toggle_str_set( std::unordered_set< std::string > &set, const std::string &str )
-{
-    auto toggled_element = std::find( set.begin(), set.end(), str );
-    if( toggled_element == set.end() ) {
-        char new_key = ' ';
-        // Find a letter in inv_chars that isn't in trait_keys.
-        for( const auto &letter : inv_chars ) {
-            bool found = false;
-            for( const auto &key : trait_keys ) {
-                if( letter == key.second ) {
-                    found = true;
-                    break;
-                }
-            }
-            if( !found ) {
-                new_key = letter;
-                break;
-            }
-        }
-        set.insert( str );
-        trait_keys[str] = new_key;
+    const auto iter = my_mutations.find( flag );
+    if( iter == my_mutations.end() ) {
+        my_mutations[flag]; // Creates a new entry with default values
     } else {
-        set.erase( toggled_element );
-        trait_keys.erase(str);
+        my_mutations.erase( iter );
     }
+    // TODO: why does this not call mutation_loss_effect / mutation_effect?
+    recalc_sight_limits();
```

Comment:
I'm thinking it's because those are handled in gaining or removing a mutation, last I knew.  Currently the only stuff that gets specifically toggled rather than mutate_towards'd is thresholds and profession traits; mutate_towards calls the effects.

So yeah, probably should get 'em for future-proofing.